src/meta/processors/admin/Balancer.cpp

```diff
@@ -206,32 +206,27 @@ Balancer::genTasks(GraphSpaceID spaceId, std::vector<HostAddr>& hostDel) {
         newHostParts.erase(h);
     }
     LOG(INFO) << "Now, try to balance the newHostParts";
+
     // We have two parts need to balance, the first one is parts on lost hosts and deleted hosts
     // The seconds one is parts on unbalanced host in newHostParts.
     std::vector<BalanceTask> tasks;
+    // 1. Iterate through all hosts that would not be included in newHostParts,
+    //    move all parts in them to host with minimum part in newHostParts
     for (auto& h : hostDel) {
         auto& lostParts = hostParts[h];
         for (auto& partId : lostParts) {
-            auto srcRet = hostWithPart(newHostParts, partId);
```

Comment:
We should keep this.  Because the deleted host maybe dead. If so, we should not choose it as "src"

Reply:
I misunderstood it as well.

In this code block, we need to move each part of each host in `hostDel` to another host. So the host in `hostDel` would be the src. In a `BalanceTask`, `src` is the host to close that part, and the one to be removed on meta. On the other hand, `dst` is the host to open the part, and the one to be added on meta.

Reply:
You are right.